src/extension.ts

```diff
@@ -53,8 +56,11 @@ export async function activateInternal(context: vscode.ExtensionContext, perfSta
     });
 
     return createApiProvider([
```

Comment:
Along with the above change making the items `readonly` on the interface, we could make this a class instance and all the fields `get` methods only.

I'm sure there are still ways around it but extra defense can't hurt.